seqr/views/apis/anvil_workspace_api.py

```diff
@@ -0,0 +1,118 @@
+"""APIs for management of projects related to AnVIL workspaces."""
+
+import logging
+import json
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import redirect
+
+from seqr.models import Project
+from seqr.views.utils.json_to_orm_utils import create_model_from_json
+from seqr.views.utils.json_utils import create_json_response
+from seqr.views.utils.file_utils import load_uploaded_file
+from seqr.views.utils.terra_api_utils import is_google_authenticated, user_get_workspace_access_level, add_service_account
+from seqr.views.apis.individual_api import _add_or_update_individuals_and_families
+from settings import API_LOGIN_REQUIRED_URL
+
+logger = logging.getLogger(__name__)
+
+
+def _workspace_can_edit(user, namespace, name):
+    if not is_google_authenticated(user):
+        return False
+    permission = user_get_workspace_access_level(user, namespace, name)
+    if not permission.get('pending') and permission.get('accessLevel') in ['WRITER', 'OWNER', 'PROJECT_OWNER']:
```

Comment:
In the function you added for adding the service account, you say that "The user must have the "can share" privilege for the workspace.". Shouldn't we check for that here?